storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+            mockedSupervisorState = conf;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorState = null;
+        }
+    }
+    private static Map mockedSupervisorState = null;
+    public static LocalState supervisorState(Map conf) throws IOException {
+        if (mockedSupervisorState != null) {
+            return null;
+        }
+        return new LocalState((supervisorLocalDir(conf) + FILE_SEPARATOR + "localstate"));
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedNimbusTopoHistoryState mocked = new SetMockedNimbusTopoHistoryState(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedNimbusTopoHistoryState. conf)]
+    //     run test ...)
+    public static class SetMockedNimbusTopoHistoryState implements Closeable {
+        public SetMockedNimbusTopoHistoryState(Map conf) {
+            mockedNimbusTopoHistoryState = conf;
+        }
+        @Override
+        public void close() {
+            mockedNimbusTopoHistoryState = null;
+        }
+    }
+    private static Map mockedNimbusTopoHistoryState = null;
+    public static LocalState nimbusTopoHistoryState(Map conf) throws IOException {
+        if (mockedNimbusTopoHistoryState != null) {
+            return null;
+        }
+        return new LocalState((masterLocalDir(conf) + FILE_SEPARATOR + "history"));
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorStormConf mocked = new SetMockedSupervisorStormConf(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorStormConf. conf)]
+    //     run test ...)
+    public static class SetMockedSupervisorStormConf implements Closeable {
+        public SetMockedSupervisorStormConf(Map conf) {
+            mockedSupervisorStormConf = conf;
+        }
+
+        @Override
+        public void close() {
+            mockedSupervisorStormConf = null;
+        }
+    }
+    private static Map mockedSupervisorStormConf = null;
+    public static Map readSupervisorStormConf(Map conf, String stormId) throws IOException {
+        if (mockedSupervisorStormConf != null) {
+            return mockedSupervisorStormConf;
+        }
+        String stormRoot = supervisorStormDistRoot(conf, stormId);
+        String confPath = supervisorStormConfPath(stormRoot);
+        return readSupervisorStormConfGivenPath(conf, confPath);
+    }
+
+    public static StormTopology readSupervisorTopology(Map conf, String stormId) throws IOException {
+        String stormRoot = supervisorStormDistRoot(conf, stormId);
+        String topologyPath = supervisorStormCodePath(stormRoot);
+        return Utils.deserialize(FileUtils.readFileToByteArray(new File(topologyPath)), StormTopology.class);
+    }
+
+    public static String workerUserRoot(Map conf) {
+        return (absoluteStormLocalDir(conf) + FILE_SEPARATOR + "/workers-users");
```

Comment:
This matches the code, but I don't think we want the extra "/"  we should file a JIRA to check on that.